rust/index/src/hnsw_provider.rs

```diff
@@ -492,10 +465,8 @@ impl ChromaError for HnswIndexProviderFlushError {
 pub enum HnswIndexProviderFileError {
     #[error("IO Error")]
     IOError(#[from] std::io::Error),
-    #[error("Storage Get Error")]
-    StorageGetError(#[from] chroma_storage::GetError),
-    #[error("Storage Put Error")]
-    StoragePutError(#[from] chroma_storage::PutError),
+    #[error("NAC Error")]
+    NACError(#[from] NetworkAdmissionControlError),
```

Comment:
I think this is another smell in the direction of "NAC" should be "AdmissionControlledStorage" and its just another Storage.